`azurerm_kubernetes_cluster`  `azurerm_kubernetes_cluster_node_pool` Fix #13919 by adding state migration to `azurerm_kubernetes_cluster` and `azurerm_kubernetes_cluster_node_pool`

lonegunmanb opened the pull request:
As #13919 said, if we import an aks and aks node pool by id like `/subscriptions/subscriptionId/resourceGroups/myResourceGroup/providers/Microsoft.ContainerService/managedClusters/myCluster`, it would caused a force new. That's because the service API returned incorrect aks id which contains `resourcegroups` instead of `resourceGroups`. When we created everything from scratch, everything would be fine, but when user imported a correct aks id, this id which contains `resourceGroups` would be set to aks id in state file, then caused this config drift.

This patch add migration functions to `azurerm_kubernetes_cluster` and `azurerm_kubernetes_cluster_node_pool` resources, which correct `azurerm_kubernetes_cluster.id` and `azurerm_kubernetes_cluster_node_pool.id` and `azurerm_kubernetes_cluster_node_pool.kubernetes_cluster_id`. Further more, this patch will set parsed cluster id to state instead of response from service API, so any new resources' id would be correct.

I've added some unit tests for migration upgrade function, here's the test results:

=== RUN   TestKubernetesClusterNodePoolV0ToV1_id
=== RUN   TestKubernetesClusterNodePoolV0ToV1_id/missing_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
=== RUN   TestKubernetesClusterNodePoolV0ToV1_id/old_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
=== RUN   TestKubernetesClusterNodePoolV0ToV1_id/new_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
--- PASS: TestKubernetesClusterNodePoolV0ToV1_id (0.00s)
    --- PASS: TestKubernetesClusterNodePoolV0ToV1_id/missing_id (0.00s)
    --- PASS: TestKubernetesClusterNodePoolV0ToV1_id/old_id (0.00s)
    --- PASS: TestKubernetesClusterNodePoolV0ToV1_id/new_id (0.00s)
=== RUN   TestKubernetesClusterNodePoolV0ToV1_kubernetes_cluster_id
=== RUN   TestKubernetesClusterNodePoolV0ToV1_kubernetes_cluster_id/missing_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
=== RUN   TestKubernetesClusterNodePoolV0ToV1_kubernetes_cluster_id/old_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
=== RUN   TestKubernetesClusterNodePoolV0ToV1_kubernetes_cluster_id/new_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
--- PASS: TestKubernetesClusterNodePoolV0ToV1_kubernetes_cluster_id (0.00s)
    --- PASS: TestKubernetesClusterNodePoolV0ToV1_kubernetes_cluster_id/missing_id (0.00s)
    --- PASS: TestKubernetesClusterNodePoolV0ToV1_kubernetes_cluster_id/old_id (0.00s)
    --- PASS: TestKubernetesClusterNodePoolV0ToV1_kubernetes_cluster_id/new_id (0.00s)
=== RUN   TestKubernetesClusterV0ToV1
=== RUN   TestKubernetesClusterV0ToV1/missing_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
=== RUN   TestKubernetesClusterV0ToV1/old_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
=== RUN   TestKubernetesClusterV0ToV1/new_id
2021/11/18 13:16:30 [DEBUG] Migrating ID to correct casing for Kubernetes Cluster
--- PASS: TestKubernetesClusterV0ToV1 (0.00s)
    --- PASS: TestKubernetesClusterV0ToV1/missing_id (0.00s)
    --- PASS: TestKubernetesClusterV0ToV1/old_id (0.00s)
    --- PASS: TestKubernetesClusterV0ToV1/new_id (0.00s)
PASS

As the Point-In-Time schema snapshot in migration function contains more than one thousand lines, it's hard for human to check whether this snapshot is equal to current schema defined in resource code, I've wrote a [helper function](https://github.com/lonegunmanb/migrationSchemaComparer) to help us check(which contains some unit test). As the schema defined in resource code would be changed in the future, we cannot just add a unit test to do this check and then push it into repository as the test would fail eventually, we can add a temporary unit test, run it, then delete it. Here's my temporary unit test added in resource code folder:

```go
package containers

import (
	"fmt"
	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/migration"
	"reflect"
	"testing"
)

func TestKubernetesClusterSchemaSnapshotV0ToV1(t *testing.T) {
	v1 := migration.KubernetesClusterV0ToV1{}
	migrationSchema := v1.Schema()
	resourceSchema := resourceKubernetesCluster().Schema
	if err := Equal(resourceSchema, migrationSchema, "root"); err != nil {
		t.Fatal(err.Error())
	}
}

func TestKubernetesClusterNodePoolSchemaSnapshotV0toV1(t *testing.T) {
	v1 := migration.KubernetesClusterNodePoolV0ToV1{}
	v1Schema := v1.Schema()
	v0Schema := resourceKubernetesClusterNodePool().Schema
	if err := Equal(v0Schema, v1Schema, "root"); err != nil {
		t.Fatal(err.Error())
	}
}

func Equal(resourceSchemas, migrationSchemas map[string]*schema.Schema, currentPath string) error {
	if len(resourceSchemas) != len(migrationSchemas) {
		return fmt.Errorf("different count, path:%s, resource:%+v, migration:%+v", currentPath, resourceSchemas, migrationSchemas)
	}
	CleanObjectValidations(resourceSchemas)
	CleanObjectValidations(migrationSchemas)
	for key, resourceSchema := range resourceSchemas {
		migrationSchema, ok := migrationSchemas[key]
		if !ok {
			return fmt.Errorf("expected %s.%s not existd in migration", currentPath, key)
		}
		if !reflect.DeepEqual(resourceSchema, migrationSchema) {
			if resourceSchema.Type == schema.TypeList || resourceSchema.Type == schema.TypeSet || resourceSchema.Type == schema.TypeMap {
				switch t := resourceSchema.Elem.(type) {
				case *schema.Resource:
					return Equal(t.Schema, migrationSchema.Elem.(*schema.Resource).Schema, fmt.Sprintf("%s.%s", currentPath, key))
				}
			}
			return fmt.Errorf("%s.%s different, resource:%+v, migration:%+v", currentPath, key, *resourceSchema, *migrationSchema)
		}
	}
	return nil
}

func CleanObjectValidations(object map[string]*schema.Schema) {
	for _, s := range object {
		CleanSchemaValidation(s)
	}
}

func CleanSchemaValidation(s *schema.Schema) {
	s.ValidateFunc = nil
	s.DiffSuppressFunc = nil
	s.StateFunc = nil
	s.ValidateDiagFunc = nil
	s.DefaultFunc = nil
	if s.Type == schema.TypeList || s.Type == schema.TypeSet || s.Type == schema.TypeMap {
		switch t := s.Elem.(type) {
		case *schema.Schema:
			CleanSchemaValidation(t)
		case *schema.Resource:
			CleanObjectValidations(t.Schema)
		}
	}
}
```

The helper functions were copied from my repository. The test results:

=== RUN   TestKubernetesClusterSchemaSnapshotV0ToV1
--- PASS: TestKubernetesClusterSchemaSnapshotV0ToV1 (0.00s)
=== RUN   TestKubernetesClusterNodePoolSchemaSnapshotV0toV1
--- PASS: TestKubernetesClusterNodePoolSchemaSnapshotV0toV1 (0.00s)
PASS

The Acc test results are here:

=== RUN   TestAccKubernetesClusterNodePool_autoScale
=== PAUSE TestAccKubernetesClusterNodePool_autoScale
=== CONT  TestAccKubernetesClusterNodePool_autoScale
--- PASS: TestAccKubernetesClusterNodePool_autoScale (1690.59s)
=== RUN   TestAccKubernetesClusterNodePool_autoScaleUpdate
=== PAUSE TestAccKubernetesClusterNodePool_autoScaleUpdate
=== CONT  TestAccKubernetesClusterNodePool_autoScaleUpdate
--- PASS: TestAccKubernetesClusterNodePool_autoScaleUpdate (1677.19s)
=== RUN   TestAccKubernetesClusterNodePool_availabilityZones
=== PAUSE TestAccKubernetesClusterNodePool_availabilityZones
=== CONT  TestAccKubernetesClusterNodePool_availabilityZones
--- PASS: TestAccKubernetesClusterNodePool_availabilityZones (1343.35s)
=== RUN   TestAccKubernetesClusterNodePool_errorForAvailabilitySet
=== PAUSE TestAccKubernetesClusterNodePool_errorForAvailabilitySet
=== CONT  TestAccKubernetesClusterNodePool_errorForAvailabilitySet
--- PASS: TestAccKubernetesClusterNodePool_errorForAvailabilitySet (865.98s)
=== RUN   TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfig
=== PAUSE TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfig
=== CONT  TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfig
--- PASS: TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfig (1266.38s)
=== RUN   TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfigPartial
=== PAUSE TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfigPartial
=== CONT  TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfigPartial
--- PASS: TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfigPartial (1209.10s)
=== RUN   TestAccKubernetesClusterNodePool_other
=== PAUSE TestAccKubernetesClusterNodePool_other
=== CONT  TestAccKubernetesClusterNodePool_other
--- PASS: TestAccKubernetesClusterNodePool_other (1472.13s)
=== RUN   TestAccKubernetesClusterNodePool_multiplePools
=== PAUSE TestAccKubernetesClusterNodePool_multiplePools
=== CONT  TestAccKubernetesClusterNodePool_multiplePools
--- PASS: TestAccKubernetesClusterNodePool_multiplePools (1629.33s)
=== RUN   TestAccKubernetesClusterNodePool_manualScale
=== PAUSE TestAccKubernetesClusterNodePool_manualScale
=== CONT  TestAccKubernetesClusterNodePool_manualScale
--- PASS: TestAccKubernetesClusterNodePool_manualScale (1292.14s)
=== RUN   TestAccKubernetesClusterNodePool_manualScaleMultiplePools
=== PAUSE TestAccKubernetesClusterNodePool_manualScaleMultiplePools
=== CONT  TestAccKubernetesClusterNodePool_manualScaleMultiplePools
--- PASS: TestAccKubernetesClusterNodePool_manualScaleMultiplePools (1390.35s)
=== RUN   TestAccKubernetesClusterNodePool_manualScaleMultiplePoolsUpdate
=== PAUSE TestAccKubernetesClusterNodePool_manualScaleMultiplePoolsUpdate
=== CONT  TestAccKubernetesClusterNodePool_manualScaleMultiplePoolsUpdate
--- PASS: TestAccKubernetesClusterNodePool_manualScaleMultiplePoolsUpdate (1693.79s)
=== RUN   TestAccKubernetesClusterNodePool_manualScaleIgnoreChanges
=== PAUSE TestAccKubernetesClusterNodePool_manualScaleIgnoreChanges
=== CONT  TestAccKubernetesClusterNodePool_manualScaleIgnoreChanges
--- PASS: TestAccKubernetesClusterNodePool_manualScaleIgnoreChanges (1445.54s)
=== RUN   TestAccKubernetesClusterNodePool_manualScaleUpdate
=== PAUSE TestAccKubernetesClusterNodePool_manualScaleUpdate
=== CONT  TestAccKubernetesClusterNodePool_manualScaleUpdate
--- PASS: TestAccKubernetesClusterNodePool_manualScaleUpdate (2076.41s)
=== RUN   TestAccKubernetesClusterNodePool_manualScaleVMSku
=== PAUSE TestAccKubernetesClusterNodePool_manualScaleVMSku
=== CONT  TestAccKubernetesClusterNodePool_manualScaleVMSku
--- PASS: TestAccKubernetesClusterNodePool_manualScaleVMSku (1755.84s)
=== RUN   TestAccKubernetesClusterNodePool_modeSystem
=== PAUSE TestAccKubernetesClusterNodePool_modeSystem
=== CONT  TestAccKubernetesClusterNodePool_modeSystem
--- PASS: TestAccKubernetesClusterNodePool_modeSystem (1426.93s)
=== RUN   TestAccKubernetesClusterNodePool_modeUpdate
=== PAUSE TestAccKubernetesClusterNodePool_modeUpdate
=== CONT  TestAccKubernetesClusterNodePool_modeUpdate
--- PASS: TestAccKubernetesClusterNodePool_modeUpdate (1746.16s)
=== RUN   TestAccKubernetesClusterNodePool_nodeLabels
=== PAUSE TestAccKubernetesClusterNodePool_nodeLabels
=== CONT  TestAccKubernetesClusterNodePool_nodeLabels
--- PASS: TestAccKubernetesClusterNodePool_nodeLabels (2123.89s)
=== RUN   TestAccKubernetesClusterNodePool_nodePublicIP
=== PAUSE TestAccKubernetesClusterNodePool_nodePublicIP
=== CONT  TestAccKubernetesClusterNodePool_nodePublicIP
--- PASS: TestAccKubernetesClusterNodePool_nodePublicIP (1244.69s)
=== RUN   TestAccKubernetesClusterNodePool_nodeTaints
=== PAUSE TestAccKubernetesClusterNodePool_nodeTaints
=== CONT  TestAccKubernetesClusterNodePool_nodeTaints
--- PASS: TestAccKubernetesClusterNodePool_nodeTaints (1309.42s)
=== RUN   TestAccKubernetesClusterNodePool_podSubnet
=== PAUSE TestAccKubernetesClusterNodePool_podSubnet
=== CONT  TestAccKubernetesClusterNodePool_podSubnet
--- PASS: TestAccKubernetesClusterNodePool_podSubnet (1455.45s)
=== RUN   TestAccKubernetesClusterNodePool_osDiskSizeGB
=== PAUSE TestAccKubernetesClusterNodePool_osDiskSizeGB
=== CONT  TestAccKubernetesClusterNodePool_osDiskSizeGB
--- PASS: TestAccKubernetesClusterNodePool_osDiskSizeGB (1198.28s)
=== RUN   TestAccKubernetesClusterNodePool_proximityPlacementGroupId
=== PAUSE TestAccKubernetesClusterNodePool_proximityPlacementGroupId
=== CONT  TestAccKubernetesClusterNodePool_proximityPlacementGroupId
--- PASS: TestAccKubernetesClusterNodePool_proximityPlacementGroupId (1230.81s)
=== RUN   TestAccKubernetesClusterNodePool_osDiskType
=== PAUSE TestAccKubernetesClusterNodePool_osDiskType
=== CONT  TestAccKubernetesClusterNodePool_osDiskType
--- PASS: TestAccKubernetesClusterNodePool_osDiskType (1201.63s)
=== RUN   TestAccKubernetesClusterNodePool_requiresImport
=== PAUSE TestAccKubernetesClusterNodePool_requiresImport
=== CONT  TestAccKubernetesClusterNodePool_requiresImport
--- PASS: TestAccKubernetesClusterNodePool_requiresImport (1223.05s)
=== RUN   TestAccKubernetesClusterNodePool_spot
=== PAUSE TestAccKubernetesClusterNodePool_spot
=== CONT  TestAccKubernetesClusterNodePool_spot
--- PASS: TestAccKubernetesClusterNodePool_spot (1146.35s)
=== RUN   TestAccKubernetesClusterNodePool_upgradeSettings
=== PAUSE TestAccKubernetesClusterNodePool_upgradeSettings
=== CONT  TestAccKubernetesClusterNodePool_upgradeSettings
--- PASS: TestAccKubernetesClusterNodePool_upgradeSettings (1752.40s)
=== RUN   TestAccKubernetesClusterNodePool_virtualNetworkAutomatic
=== PAUSE TestAccKubernetesClusterNodePool_virtualNetworkAutomatic
=== CONT  TestAccKubernetesClusterNodePool_virtualNetworkAutomatic
--- PASS: TestAccKubernetesClusterNodePool_virtualNetworkAutomatic (1282.61s)
=== RUN   TestAccKubernetesClusterNodePool_virtualNetworkManual
=== PAUSE TestAccKubernetesClusterNodePool_virtualNetworkManual
=== CONT  TestAccKubernetesClusterNodePool_virtualNetworkManual
--- PASS: TestAccKubernetesClusterNodePool_virtualNetworkManual (1344.51s)
=== RUN   TestAccKubernetesClusterNodePool_windows
=== PAUSE TestAccKubernetesClusterNodePool_windows
=== CONT  TestAccKubernetesClusterNodePool_windows
--- PASS: TestAccKubernetesClusterNodePool_windows (1506.26s)
=== RUN   TestAccKubernetesClusterNodePool_windowsAndLinux
=== PAUSE TestAccKubernetesClusterNodePool_windowsAndLinux
=== CONT  TestAccKubernetesClusterNodePool_windowsAndLinux
--- PASS: TestAccKubernetesClusterNodePool_windowsAndLinux (1552.75s)
=== RUN   TestAccKubernetesClusterNodePool_zeroSize
=== PAUSE TestAccKubernetesClusterNodePool_zeroSize
=== CONT  TestAccKubernetesClusterNodePool_zeroSize
--- PASS: TestAccKubernetesClusterNodePool_zeroSize (988.16s)
=== RUN   TestAccKubernetesClusterNodePool_hostEncryption
=== PAUSE TestAccKubernetesClusterNodePool_hostEncryption
=== CONT  TestAccKubernetesClusterNodePool_hostEncryption
--- PASS: TestAccKubernetesClusterNodePool_hostEncryption (1245.28s)
=== RUN   TestAccKubernetesClusterNodePool_maxSize
=== PAUSE TestAccKubernetesClusterNodePool_maxSize
=== CONT  TestAccKubernetesClusterNodePool_maxSize
--- PASS: TestAccKubernetesClusterNodePool_maxSize (1228.43s)
=== RUN   TestAccKubernetesClusterNodePool_sameSize
=== PAUSE TestAccKubernetesClusterNodePool_sameSize
=== CONT  TestAccKubernetesClusterNodePool_sameSize
--- PASS: TestAccKubernetesClusterNodePool_sameSize (1228.87s)
=== RUN   TestAccKubernetesClusterNodePool_ultraSSD
=== PAUSE TestAccKubernetesClusterNodePool_ultraSSD
=== CONT  TestAccKubernetesClusterNodePool_ultraSSD
--- PASS: TestAccKubernetesClusterNodePool_ultraSSD (1280.45s)
=== RUN   TestAccKubernetesClusterNodePool_osSku
=== PAUSE TestAccKubernetesClusterNodePool_osSku
=== CONT  TestAccKubernetesClusterNodePool_osSku
--- PASS: TestAccKubernetesClusterNodePool_osSku (1009.85s)
=== RUN   TestAccKubernetesClusterNodePool_turnOnEnableAutoScalingWithDefaultMaxMinCountSettings
=== PAUSE TestAccKubernetesClusterNodePool_turnOnEnableAutoScalingWithDefaultMaxMinCountSettings
=== CONT  TestAccKubernetesClusterNodePool_turnOnEnableAutoScalingWithDefaultMaxMinCountSettings
--- PASS: TestAccKubernetesClusterNodePool_turnOnEnableAutoScalingWithDefaultMaxMinCountSettings (1258.31s)
PASS

=== RUN   TestAccKubernetesCluster_addonProfileAciConnectorLinux
=== PAUSE TestAccKubernetesCluster_addonProfileAciConnectorLinux
=== CONT  TestAccKubernetesCluster_addonProfileAciConnectorLinux
--- PASS: TestAccKubernetesCluster_addonProfileAciConnectorLinux (1221.23s)
=== RUN   TestAccKubernetesCluster_addonProfileAciConnectorLinuxDisabled
=== PAUSE TestAccKubernetesCluster_addonProfileAciConnectorLinuxDisabled
=== CONT  TestAccKubernetesCluster_addonProfileAciConnectorLinuxDisabled
--- PASS: TestAccKubernetesCluster_addonProfileAciConnectorLinuxDisabled (1075.80s)
=== RUN   TestAccKubernetesCluster_addonProfileAzurePolicy
=== PAUSE TestAccKubernetesCluster_addonProfileAzurePolicy
=== CONT  TestAccKubernetesCluster_addonProfileAzurePolicy
--- PASS: TestAccKubernetesCluster_addonProfileAzurePolicy (1412.18s)
=== RUN   TestAccKubernetesCluster_addonProfileKubeDashboard
=== PAUSE TestAccKubernetesCluster_addonProfileKubeDashboard
=== CONT  TestAccKubernetesCluster_addonProfileKubeDashboard
--- PASS: TestAccKubernetesCluster_addonProfileKubeDashboard (899.99s)
=== RUN   TestAccKubernetesCluster_addonProfileOMS
=== PAUSE TestAccKubernetesCluster_addonProfileOMS
=== CONT  TestAccKubernetesCluster_addonProfileOMS
--- PASS: TestAccKubernetesCluster_addonProfileOMS (991.97s)
=== RUN   TestAccKubernetesCluster_addonProfileOMSToggle
=== PAUSE TestAccKubernetesCluster_addonProfileOMSToggle
=== CONT  TestAccKubernetesCluster_addonProfileOMSToggle
--- PASS: TestAccKubernetesCluster_addonProfileOMSToggle (2118.67s)
=== RUN   TestAccKubernetesCluster_addonProfileRoutingToggle
=== PAUSE TestAccKubernetesCluster_addonProfileRoutingToggle
=== CONT  TestAccKubernetesCluster_addonProfileRoutingToggle
--- PASS: TestAccKubernetesCluster_addonProfileRoutingToggle (1279.40s)
=== RUN   TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId
=== PAUSE TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId
=== CONT  TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId
--- PASS: TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId (1657.39s)
=== RUN   TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetCIDR
=== PAUSE TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetCIDR
=== CONT  TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetCIDR
--- PASS: TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetCIDR (1206.08s)
=== RUN   TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetId
=== PAUSE TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetId
=== CONT  TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetId
--- PASS: TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetId (1316.13s)
=== RUN   TestAccKubernetesCluster_addonProfileOpenServiceMesh
=== PAUSE TestAccKubernetesCluster_addonProfileOpenServiceMesh
=== CONT  TestAccKubernetesCluster_addonProfileOpenServiceMesh
--- PASS: TestAccKubernetesCluster_addonProfileOpenServiceMesh (964.98s)
=== RUN   TestAccKubernetesCluster_apiServerAuthorizedIPRanges
=== PAUSE TestAccKubernetesCluster_apiServerAuthorizedIPRanges
=== CONT  TestAccKubernetesCluster_apiServerAuthorizedIPRanges
--- PASS: TestAccKubernetesCluster_apiServerAuthorizedIPRanges (973.87s)
=== RUN   TestAccKubernetesCluster_managedClusterIdentity
=== PAUSE TestAccKubernetesCluster_managedClusterIdentity
=== CONT  TestAccKubernetesCluster_managedClusterIdentity
--- PASS: TestAccKubernetesCluster_managedClusterIdentity (967.90s)
=== RUN   TestAccKubernetesCluster_userAssignedIdentity
=== PAUSE TestAccKubernetesCluster_userAssignedIdentity
=== CONT  TestAccKubernetesCluster_userAssignedIdentity
--- PASS: TestAccKubernetesCluster_userAssignedIdentity (986.22s)
=== RUN   TestAccKubernetesCluster_updateWithUserAssignedIdentity
=== PAUSE TestAccKubernetesCluster_updateWithUserAssignedIdentity
=== CONT  TestAccKubernetesCluster_updateWithUserAssignedIdentity
--- PASS: TestAccKubernetesCluster_updateWithUserAssignedIdentity (1192.05s)
=== RUN   TestAccKubernetesCluster_userAssignedKubeletIdentity
=== PAUSE TestAccKubernetesCluster_userAssignedKubeletIdentity
=== CONT  TestAccKubernetesCluster_userAssignedKubeletIdentity
--- PASS: TestAccKubernetesCluster_userAssignedKubeletIdentity (982.31s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControl
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControl
=== CONT  TestAccKubernetesCluster_roleBasedAccessControl
--- PASS: TestAccKubernetesCluster_roleBasedAccessControl (991.85s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControlAAD
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControlAAD
=== CONT  TestAccKubernetesCluster_roleBasedAccessControlAAD
--- PASS: TestAccKubernetesCluster_roleBasedAccessControlAAD (655.20s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControlAADUpdateToManaged
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControlAADUpdateToManaged
=== CONT  TestAccKubernetesCluster_roleBasedAccessControlAADUpdateToManaged
--- PASS: TestAccKubernetesCluster_roleBasedAccessControlAADUpdateToManaged (866.43s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControlAADUpdateToManagedSensitive
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControlAADUpdateToManagedSensitive
=== CONT  TestAccKubernetesCluster_roleBasedAccessControlAADUpdateToManagedSensitive
--- PASS: TestAccKubernetesCluster_roleBasedAccessControlAADUpdateToManagedSensitive (768.25s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControlAADManaged
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControlAADManaged
=== CONT  TestAccKubernetesCluster_roleBasedAccessControlAADManaged
--- PASS: TestAccKubernetesCluster_roleBasedAccessControlAADManaged (1023.49s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControlAADManagedWithLocalAccountDisabled
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControlAADManagedWithLocalAccountDisabled
=== CONT  TestAccKubernetesCluster_roleBasedAccessControlAADManagedWithLocalAccountDisabled
--- PASS: TestAccKubernetesCluster_roleBasedAccessControlAADManagedWithLocalAccountDisabled (905.58s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControlAADManagedWithLocalAccountDisabledUpdated
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControlAADManagedWithLocalAccountDisabledUpdated
=== CONT  TestAccKubernetesCluster_roleBasedAccessControlAADManagedWithLocalAccountDisabledUpdated
--- PASS: TestAccKubernetesCluster_roleBasedAccessControlAADManagedWithLocalAccountDisabledUpdated (1351.92s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControlAADManagedChange
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControlAADManagedChange
=== CONT  TestAccKubernetesCluster_roleBasedAccessControlAADManagedChange
--- PASS: TestAccKubernetesCluster_roleBasedAccessControlAADManagedChange (1318.73s)
=== RUN   TestAccKubernetesCluster_roleBasedAccessControlAzure
=== PAUSE TestAccKubernetesCluster_roleBasedAccessControlAzure
=== CONT  TestAccKubernetesCluster_roleBasedAccessControlAzure
--- PASS: TestAccKubernetesCluster_roleBasedAccessControlAzure (1383.74s)
=== RUN   TestAccKubernetesCluster_servicePrincipal
=== PAUSE TestAccKubernetesCluster_servicePrincipal
=== CONT  TestAccKubernetesCluster_servicePrincipal
--- PASS: TestAccKubernetesCluster_servicePrincipal (1168.23s)
=== RUN   TestAccKubernetesCluster_servicePrincipalToSystemAssignedIdentity
=== PAUSE TestAccKubernetesCluster_servicePrincipalToSystemAssignedIdentity
=== CONT  TestAccKubernetesCluster_servicePrincipalToSystemAssignedIdentity
--- PASS: TestAccKubernetesCluster_servicePrincipalToSystemAssignedIdentity (1127.98s)
=== RUN   TestAccKubernetesCluster_servicePrincipalToUserAssignedIdentity
=== PAUSE TestAccKubernetesCluster_servicePrincipalToUserAssignedIdentity
=== CONT  TestAccKubernetesCluster_servicePrincipalToUserAssignedIdentity
--- PASS: TestAccKubernetesCluster_servicePrincipalToUserAssignedIdentity (1361.49s)
=== RUN   TestAccKubernetesCluster_updateRoleBasedAccessControlAAD
=== PAUSE TestAccKubernetesCluster_updateRoleBasedAccessControlAAD
=== CONT  TestAccKubernetesCluster_updateRoleBasedAccessControlAAD
--- PASS: TestAccKubernetesCluster_updateRoleBasedAccessControlAAD (811.89s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingKubenet
=== PAUSE TestAccKubernetesCluster_advancedNetworkingKubenet
=== CONT  TestAccKubernetesCluster_advancedNetworkingKubenet
--- PASS: TestAccKubernetesCluster_advancedNetworkingKubenet (1093.72s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingKubenetComplete
=== PAUSE TestAccKubernetesCluster_advancedNetworkingKubenetComplete
=== CONT  TestAccKubernetesCluster_advancedNetworkingKubenetComplete
--- PASS: TestAccKubernetesCluster_advancedNetworkingKubenetComplete (1072.97s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingAzure
=== PAUSE TestAccKubernetesCluster_advancedNetworkingAzure
=== CONT  TestAccKubernetesCluster_advancedNetworkingAzure
--- PASS: TestAccKubernetesCluster_advancedNetworkingAzure (1012.60s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingAzureComplete
=== PAUSE TestAccKubernetesCluster_advancedNetworkingAzureComplete
=== CONT  TestAccKubernetesCluster_advancedNetworkingAzureComplete
--- PASS: TestAccKubernetesCluster_advancedNetworkingAzureComplete (1103.55s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicy
=== PAUSE TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicy
=== CONT  TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicy
--- PASS: TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicy (1158.87s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicyComplete
=== PAUSE TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicyComplete
=== CONT  TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicyComplete
--- PASS: TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicyComplete (1218.91s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicyNetworkModeTransparent
=== PAUSE TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicyNetworkModeTransparent
=== CONT  TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicyNetworkModeTransparent
--- PASS: TestAccKubernetesCluster_advancedNetworkingAzureCalicoPolicyNetworkModeTransparent (1228.09s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingAzureNPMPolicy
=== PAUSE TestAccKubernetesCluster_advancedNetworkingAzureNPMPolicy
=== CONT  TestAccKubernetesCluster_advancedNetworkingAzureNPMPolicy
--- PASS: TestAccKubernetesCluster_advancedNetworkingAzureNPMPolicy (1102.50s)
=== RUN   TestAccKubernetesCluster_advancedNetworkingAzureNPMPolicyComplete
=== PAUSE TestAccKubernetesCluster_advancedNetworkingAzureNPMPolicyComplete
=== CONT  TestAccKubernetesCluster_advancedNetworkingAzureNPMPolicyComplete
--- PASS: TestAccKubernetesCluster_advancedNetworkingAzureNPMPolicyComplete (1171.73s)
=== RUN   TestAccKubernetesCluster_enableNodePublicIP
=== PAUSE TestAccKubernetesCluster_enableNodePublicIP
=== CONT  TestAccKubernetesCluster_enableNodePublicIP
--- PASS: TestAccKubernetesCluster_enableNodePublicIP (1180.76s)
=== RUN   TestAccKubernetesCluster_internalNetwork
=== PAUSE TestAccKubernetesCluster_internalNetwork
=== CONT  TestAccKubernetesCluster_internalNetwork
--- PASS: TestAccKubernetesCluster_internalNetwork (1165.52s)
=== RUN   TestAccKubernetesCluster_nodePublicIPPrefix
=== PAUSE TestAccKubernetesCluster_nodePublicIPPrefix
=== CONT  TestAccKubernetesCluster_nodePublicIPPrefix
--- PASS: TestAccKubernetesCluster_nodePublicIPPrefix (961.23s)
=== RUN   TestAccKubernetesCluster_outboundTypeLoadBalancer
=== PAUSE TestAccKubernetesCluster_outboundTypeLoadBalancer
=== CONT  TestAccKubernetesCluster_outboundTypeLoadBalancer
--- PASS: TestAccKubernetesCluster_outboundTypeLoadBalancer (971.09s)
=== RUN   TestAccKubernetesCluster_privateClusterOn
=== PAUSE TestAccKubernetesCluster_privateClusterOn
=== CONT  TestAccKubernetesCluster_privateClusterOn
--- PASS: TestAccKubernetesCluster_privateClusterOn (1129.76s)
=== RUN   TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZone
=== PAUSE TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZone
=== CONT  TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZone
--- PASS: TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZone (1207.29s)
=== RUN   TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneAndServicePrincipal
=== PAUSE TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneAndServicePrincipal
=== CONT  TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneAndServicePrincipal
--- PASS: TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneAndServicePrincipal (1068.48s)
=== RUN   TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSubDomain
=== PAUSE TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSubDomain
=== CONT  TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSubDomain
--- PASS: TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSubDomain (1184.09s)
=== RUN   TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSystem
=== PAUSE TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSystem
=== CONT  TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSystem
--- PASS: TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSystem (1187.66s)
=== RUN   TestAccKubernetesCluster_privateClusterOff
=== PAUSE TestAccKubernetesCluster_privateClusterOff
=== CONT  TestAccKubernetesCluster_privateClusterOff
--- PASS: TestAccKubernetesCluster_privateClusterOff (1082.70s)
=== RUN   TestAccKubernetesCluster_standardLoadBalancer
=== PAUSE TestAccKubernetesCluster_standardLoadBalancer
=== CONT  TestAccKubernetesCluster_standardLoadBalancer
--- PASS: TestAccKubernetesCluster_standardLoadBalancer (1181.34s)
=== RUN   TestAccKubernetesCluster_standardLoadBalancerComplete
=== PAUSE TestAccKubernetesCluster_standardLoadBalancerComplete
=== CONT  TestAccKubernetesCluster_standardLoadBalancerComplete
--- PASS: TestAccKubernetesCluster_standardLoadBalancerComplete (1130.26s)
=== RUN   TestAccKubernetesCluster_standardLoadBalancerProfile
=== PAUSE TestAccKubernetesCluster_standardLoadBalancerProfile
=== CONT  TestAccKubernetesCluster_standardLoadBalancerProfile
--- PASS: TestAccKubernetesCluster_standardLoadBalancerProfile (1018.16s)
=== RUN   TestAccKubernetesCluster_standardLoadBalancerProfileComplete
=== PAUSE TestAccKubernetesCluster_standardLoadBalancerProfileComplete
=== CONT  TestAccKubernetesCluster_standardLoadBalancerProfileComplete
--- PASS: TestAccKubernetesCluster_standardLoadBalancerProfileComplete (1140.96s)
=== RUN   TestAccKubernetesCluster_standardLoadBalancerProfileWithPortAndTimeout
=== PAUSE TestAccKubernetesCluster_standardLoadBalancerProfileWithPortAndTimeout
=== CONT  TestAccKubernetesCluster_standardLoadBalancerProfileWithPortAndTimeout
--- PASS: TestAccKubernetesCluster_standardLoadBalancerProfileWithPortAndTimeout (939.40s)
=== RUN   TestAccKubernetesCluster_basicLoadBalancerProfile
=== PAUSE TestAccKubernetesCluster_basicLoadBalancerProfile
=== CONT  TestAccKubernetesCluster_basicLoadBalancerProfile
--- PASS: TestAccKubernetesCluster_basicLoadBalancerProfile (238.41s)
=== RUN   TestAccKubernetesCluster_prefixedLoadBalancerProfile
=== PAUSE TestAccKubernetesCluster_prefixedLoadBalancerProfile
=== CONT  TestAccKubernetesCluster_prefixedLoadBalancerProfile
--- PASS: TestAccKubernetesCluster_prefixedLoadBalancerProfile (1073.55s)
=== RUN   TestAccKubernetesCluster_changingLoadBalancerProfile
=== PAUSE TestAccKubernetesCluster_changingLoadBalancerProfile
=== CONT  TestAccKubernetesCluster_changingLoadBalancerProfile
--- PASS: TestAccKubernetesCluster_changingLoadBalancerProfile (1819.10s)
=== RUN   TestAccKubernetesCluster_basicAvailabilitySet
=== PAUSE TestAccKubernetesCluster_basicAvailabilitySet
=== CONT  TestAccKubernetesCluster_basicAvailabilitySet
--- PASS: TestAccKubernetesCluster_basicAvailabilitySet (871.29s)
=== RUN   TestAccKubernetesCluster_sameSize
=== PAUSE TestAccKubernetesCluster_sameSize
=== CONT  TestAccKubernetesCluster_sameSize
--- PASS: TestAccKubernetesCluster_sameSize (878.18s)
=== RUN   TestAccKubernetesCluster_basicAvailabilitySetSensitive
=== PAUSE TestAccKubernetesCluster_basicAvailabilitySetSensitive
=== CONT  TestAccKubernetesCluster_basicAvailabilitySetSensitive
--- PASS: TestAccKubernetesCluster_basicAvailabilitySetSensitive (997.77s)
=== RUN   TestAccKubernetesCluster_basicVMSS
=== PAUSE TestAccKubernetesCluster_basicVMSS
=== CONT  TestAccKubernetesCluster_basicVMSS
--- PASS: TestAccKubernetesCluster_basicVMSS (903.59s)
=== RUN   TestAccKubernetesCluster_requiresImport
=== PAUSE TestAccKubernetesCluster_requiresImport
=== CONT  TestAccKubernetesCluster_requiresImport
--- PASS: TestAccKubernetesCluster_requiresImport (896.89s)
=== RUN   TestAccKubernetesCluster_criticalAddonsTaint
=== PAUSE TestAccKubernetesCluster_criticalAddonsTaint
=== CONT  TestAccKubernetesCluster_criticalAddonsTaint
--- PASS: TestAccKubernetesCluster_criticalAddonsTaint (904.45s)
=== RUN   TestAccKubernetesCluster_kubeletAndLinuxOSConfig
=== PAUSE TestAccKubernetesCluster_kubeletAndLinuxOSConfig
=== CONT  TestAccKubernetesCluster_kubeletAndLinuxOSConfig
--- PASS: TestAccKubernetesCluster_kubeletAndLinuxOSConfig (951.12s)
=== RUN   TestAccKubernetesCluster_kubeletAndLinuxOSConfigPartial
=== PAUSE TestAccKubernetesCluster_kubeletAndLinuxOSConfigPartial
=== CONT  TestAccKubernetesCluster_kubeletAndLinuxOSConfigPartial
--- PASS: TestAccKubernetesCluster_kubeletAndLinuxOSConfigPartial (860.21s)
=== RUN   TestAccKubernetesCluster_linuxProfile
=== PAUSE TestAccKubernetesCluster_linuxProfile
=== CONT  TestAccKubernetesCluster_linuxProfile
--- PASS: TestAccKubernetesCluster_linuxProfile (926.81s)
=== RUN   TestAccKubernetesCluster_nodeLabels
=== PAUSE TestAccKubernetesCluster_nodeLabels
=== CONT  TestAccKubernetesCluster_nodeLabels
--- PASS: TestAccKubernetesCluster_nodeLabels (2372.08s)
=== RUN   TestAccKubernetesCluster_nodeResourceGroup
=== PAUSE TestAccKubernetesCluster_nodeResourceGroup
=== CONT  TestAccKubernetesCluster_nodeResourceGroup
--- PASS: TestAccKubernetesCluster_nodeResourceGroup (870.43s)
=== RUN   TestAccKubernetesCluster_nodePoolOther
=== PAUSE TestAccKubernetesCluster_nodePoolOther
=== CONT  TestAccKubernetesCluster_nodePoolOther
--- PASS: TestAccKubernetesCluster_nodePoolOther (977.66s)
=== RUN   TestAccKubernetesCluster_upgradeSkuTier
=== PAUSE TestAccKubernetesCluster_upgradeSkuTier
=== CONT  TestAccKubernetesCluster_upgradeSkuTier
--- PASS: TestAccKubernetesCluster_upgradeSkuTier (1394.22s)
=== RUN   TestAccKubernetesCluster_paidSku
=== PAUSE TestAccKubernetesCluster_paidSku
=== CONT  TestAccKubernetesCluster_paidSku
--- PASS: TestAccKubernetesCluster_paidSku (939.06s)
=== RUN   TestAccKubernetesCluster_podSubnet
=== PAUSE TestAccKubernetesCluster_podSubnet
=== CONT  TestAccKubernetesCluster_podSubnet
--- PASS: TestAccKubernetesCluster_podSubnet (1119.77s)
=== RUN   TestAccKubernetesCluster_upgrade
=== PAUSE TestAccKubernetesCluster_upgrade
=== CONT  TestAccKubernetesCluster_upgrade
--- PASS: TestAccKubernetesCluster_upgrade (1205.00s)
=== RUN   TestAccKubernetesCluster_tags
=== PAUSE TestAccKubernetesCluster_tags
=== CONT  TestAccKubernetesCluster_tags
--- PASS: TestAccKubernetesCluster_tags (1207.27s)
=== RUN   TestAccKubernetesCluster_windowsProfile
=== PAUSE TestAccKubernetesCluster_windowsProfile
=== CONT  TestAccKubernetesCluster_windowsProfile
--- PASS: TestAccKubernetesCluster_windowsProfile (869.35s)
=== RUN   TestAccKubernetesCluster_windowsProfileLicense
=== PAUSE TestAccKubernetesCluster_windowsProfileLicense
=== CONT  TestAccKubernetesCluster_windowsProfileLicense
--- PASS: TestAccKubernetesCluster_windowsProfileLicense (938.29s)
=== RUN   TestAccKubernetesCluster_updateWindowsProfileLicense
=== PAUSE TestAccKubernetesCluster_updateWindowsProfileLicense
=== CONT  TestAccKubernetesCluster_updateWindowsProfileLicense
--- PASS: TestAccKubernetesCluster_updateWindowsProfileLicense (1406.57s)
=== RUN   TestAccKubernetesCluster_diskEncryption
=== PAUSE TestAccKubernetesCluster_diskEncryption
=== CONT  TestAccKubernetesCluster_diskEncryption
--- PASS: TestAccKubernetesCluster_diskEncryption (1218.75s)
=== RUN   TestAccKubernetesCluster_upgradeChannel
=== PAUSE TestAccKubernetesCluster_upgradeChannel
=== CONT  TestAccKubernetesCluster_upgradeChannel
--- PASS: TestAccKubernetesCluster_upgradeChannel (2032.50s)
=== RUN   TestAccKubernetesCluster_basicMaintenanceConfig
=== PAUSE TestAccKubernetesCluster_basicMaintenanceConfig
=== CONT  TestAccKubernetesCluster_basicMaintenanceConfig
--- PASS: TestAccKubernetesCluster_basicMaintenanceConfig (871.00s)
=== RUN   TestAccKubernetesCluster_completeMaintenanceConfig
=== PAUSE TestAccKubernetesCluster_completeMaintenanceConfig
=== CONT  TestAccKubernetesCluster_completeMaintenanceConfig
--- PASS: TestAccKubernetesCluster_completeMaintenanceConfig (897.11s)
=== RUN   TestAccKubernetesCluster_updateMaintenanceConfig
=== PAUSE TestAccKubernetesCluster_updateMaintenanceConfig
=== CONT  TestAccKubernetesCluster_updateMaintenanceConfig
--- PASS: TestAccKubernetesCluster_updateMaintenanceConfig (1423.42s)
=== RUN   TestAccKubernetesCluster_ultraSSD
=== PAUSE TestAccKubernetesCluster_ultraSSD
=== CONT  TestAccKubernetesCluster_ultraSSD
=== CONT  TestAccKubernetesCluster_ultraSSD
    testcase.go:110: Step 5/6 error: Error running apply: exit status 1
        
        Error: waiting for creation of Managed Kubernetes Cluster "acctestaks211117214102337711" (Resource Group "acctestRG-aks-211117214102337711"): Code="ReconcileMSICredentialError" Message="AKS encountered an internal error while attempting the requested Creating operation. AKS will continuously retry the requested operation until successful or a retry timeout is hit. Check back to see if the operation requires resubmission. Correlation ID: 4aefc913-7075-31e0-ce3e-1502cd0431be, Operation ID: 8e6315d3-d705-4cf2-9613-0cdf6f7782e2, Timestamp: 2021-11-17T16:00:43Z."
        
          with azurerm_kubernetes_cluster.test,
          on terraform_plugin_test.tf line 9, in resource "azurerm_kubernetes_cluster" "test":
           9: resource "azurerm_kubernetes_cluster" "test" {
        
--- FAIL: TestAccKubernetesCluster_ultraSSD (1936.77s)

=== RUN   TestAccKubernetesCluster_privateClusterPublicFqdn
=== PAUSE TestAccKubernetesCluster_privateClusterPublicFqdn
=== CONT  TestAccKubernetesCluster_privateClusterPublicFqdn
--- PASS: TestAccKubernetesCluster_privateClusterPublicFqdn (1423.72s)
=== RUN   TestAccKubernetesCluster_osSku
=== PAUSE TestAccKubernetesCluster_osSku
=== CONT  TestAccKubernetesCluster_osSku
--- PASS: TestAccKubernetesCluster_osSku (936.02s)
=== RUN   TestAccKubernetesCluster_hostEncryption
=== PAUSE TestAccKubernetesCluster_hostEncryption
=== CONT  TestAccKubernetesCluster_hostEncryption
--- PASS: TestAccKubernetesCluster_hostEncryption (891.34s)
=== RUN   TestAccKubernetesCluster_addAgent
=== PAUSE TestAccKubernetesCluster_addAgent
=== CONT  TestAccKubernetesCluster_addAgent
--- PASS: TestAccKubernetesCluster_addAgent (1527.84s)
=== RUN   TestAccKubernetesCluster_manualScaleIgnoreChanges
=== PAUSE TestAccKubernetesCluster_manualScaleIgnoreChanges
=== CONT  TestAccKubernetesCluster_manualScaleIgnoreChanges
--- PASS: TestAccKubernetesCluster_manualScaleIgnoreChanges (1225.84s)
=== RUN   TestAccKubernetesCluster_removeAgent
=== PAUSE TestAccKubernetesCluster_removeAgent
=== CONT  TestAccKubernetesCluster_removeAgent
--- PASS: TestAccKubernetesCluster_removeAgent (1243.50s)
=== RUN   TestAccKubernetesCluster_autoScalingError
=== PAUSE TestAccKubernetesCluster_autoScalingError
=== CONT  TestAccKubernetesCluster_autoScalingError
--- PASS: TestAccKubernetesCluster_autoScalingError (996.85s)
=== RUN   TestAccKubernetesCluster_autoScalingErrorMax
=== PAUSE TestAccKubernetesCluster_autoScalingErrorMax
=== CONT  TestAccKubernetesCluster_autoScalingErrorMax
--- PASS: TestAccKubernetesCluster_autoScalingErrorMax (190.45s)
=== RUN   TestAccKubernetesCluster_autoScalingWithMaxCount
=== PAUSE TestAccKubernetesCluster_autoScalingWithMaxCount
=== CONT  TestAccKubernetesCluster_autoScalingWithMaxCount
--- PASS: TestAccKubernetesCluster_autoScalingWithMaxCount (946.97s)
=== RUN   TestAccKubernetesCluster_autoScalingErrorMin
=== PAUSE TestAccKubernetesCluster_autoScalingErrorMin
=== CONT  TestAccKubernetesCluster_autoScalingErrorMin
--- PASS: TestAccKubernetesCluster_autoScalingErrorMin (198.04s)
=== RUN   TestAccKubernetesCluster_autoScalingNodeCountUnset
=== PAUSE TestAccKubernetesCluster_autoScalingNodeCountUnset
=== CONT  TestAccKubernetesCluster_autoScalingNodeCountUnset
--- PASS: TestAccKubernetesCluster_autoScalingNodeCountUnset (887.69s)
=== RUN   TestAccKubernetesCluster_autoScalingNoAvailabilityZones
=== PAUSE TestAccKubernetesCluster_autoScalingNoAvailabilityZones
=== CONT  TestAccKubernetesCluster_autoScalingNoAvailabilityZones
--- PASS: TestAccKubernetesCluster_autoScalingNoAvailabilityZones (932.15s)
=== RUN   TestAccKubernetesCluster_autoScalingWithAvailabilityZones
=== PAUSE TestAccKubernetesCluster_autoScalingWithAvailabilityZones
=== CONT  TestAccKubernetesCluster_autoScalingWithAvailabilityZones
--- PASS: TestAccKubernetesCluster_autoScalingWithAvailabilityZones (910.73s)
=== RUN   TestAccKubernetesCluster_autoScalingProfile
=== PAUSE TestAccKubernetesCluster_autoScalingProfile
=== CONT  TestAccKubernetesCluster_autoScalingProfile
--- PASS: TestAccKubernetesCluster_autoScalingProfile (935.33s)
=== RUN   TestAccKubernetesCluster_upgradeAutoScaleMinCount
=== PAUSE TestAccKubernetesCluster_upgradeAutoScaleMinCount
=== CONT  TestAccKubernetesCluster_upgradeAutoScaleMinCount
--- PASS: TestAccKubernetesCluster_upgradeAutoScaleMinCount (1274.70s)
=== RUN   TestAccKubernetesCluster_upgradeControlPlane
=== PAUSE TestAccKubernetesCluster_upgradeControlPlane
=== CONT  TestAccKubernetesCluster_upgradeControlPlane
--- PASS: TestAccKubernetesCluster_upgradeControlPlane (1192.37s)
=== RUN   TestAccKubernetesCluster_upgradeControlPlaneAndDefaultNodePoolTogether
=== PAUSE TestAccKubernetesCluster_upgradeControlPlaneAndDefaultNodePoolTogether
=== CONT  TestAccKubernetesCluster_upgradeControlPlaneAndDefaultNodePoolTogether
--- PASS: TestAccKubernetesCluster_upgradeControlPlaneAndDefaultNodePoolTogether (1776.82s)
=== RUN   TestAccKubernetesCluster_upgradeControlPlaneAndDefaultNodePoolTwoPhase
=== PAUSE TestAccKubernetesCluster_upgradeControlPlaneAndDefaultNodePoolTwoPhase
=== CONT  TestAccKubernetesCluster_upgradeControlPlaneAndDefaultNodePoolTwoPhase
--- PASS: TestAccKubernetesCluster_upgradeControlPlaneAndDefaultNodePoolTwoPhase (1775.94s)
=== RUN   TestAccKubernetesCluster_upgradeNodePoolBeforeControlPlaneFails
=== PAUSE TestAccKubernetesCluster_upgradeNodePoolBeforeControlPlaneFails
=== CONT  TestAccKubernetesCluster_upgradeNodePoolBeforeControlPlaneFails
--- PASS: TestAccKubernetesCluster_upgradeNodePoolBeforeControlPlaneFails (1134.44s)
=== RUN   TestAccKubernetesCluster_upgradeCustomNodePoolAfterControlPlane
=== PAUSE TestAccKubernetesCluster_upgradeCustomNodePoolAfterControlPlane
=== CONT  TestAccKubernetesCluster_upgradeCustomNodePoolAfterControlPlane
--- PASS: TestAccKubernetesCluster_upgradeCustomNodePoolAfterControlPlane (2304.55s)
=== RUN   TestAccKubernetesCluster_upgradeCustomNodePoolBeforeControlPlaneFails
=== PAUSE TestAccKubernetesCluster_upgradeCustomNodePoolBeforeControlPlaneFails
=== CONT  TestAccKubernetesCluster_upgradeCustomNodePoolBeforeControlPlaneFails
--- PASS: TestAccKubernetesCluster_upgradeCustomNodePoolBeforeControlPlaneFails (1364.80s)
=== RUN   TestAccKubernetesCluster_upgradeSettings
=== PAUSE TestAccKubernetesCluster_upgradeSettings
=== CONT  TestAccKubernetesCluster_upgradeSettings
--- PASS: TestAccKubernetesCluster_upgradeSettings (1463.98s)
FAIL

Process finished with the exit code 1

The only one failed test also failed in main branch, so I guess it's ok for this patch.

stephybun commented on internal/services/containers/migration/kubernetes_cluster.go:
```diff
@@ -0,0 +1,1421 @@
+		"location": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			StateFunc:        location.StateFunc,
+			DiffSuppressFunc: location.DiffSuppressFunc,
```
State migrations only require the Type and Required/Optional/Computed for most fields, things like List of Strings also require the Elem Type, so these fields can be removed everywhere in `Schema()`

lonegunmanb replied:
Copy that, I'll remove these functions, thanks for the tip!

stephybun reviewed (changes requested):
Thanks for this PR @lonegunmanb. Overall this looks good, I left a comment regarding the Schemas for the state migrations.

lonegunmanb commented:
I've removed fields except Type/Required/Optional/Computed from schema snapshot, and updated my [check tool](https://github.com/lonegunmanb/migrationSchemaComparer), the unit test show passed.
Since I kept code other than migration schema untouched, I didn't run acc tests this time.

tombuildsstuff reviewed (changes requested):
hi @lonegunmanb 

Thanks for this PR.

I've taken a look through and left a couple of comments inline, but rather than looking up and parsing the Azure ID here, we can instead omit the `Get` request and use the `New{Name}ID` functions here both for setting the Resource ID and for the Requires Import check here - would you be able to take a look?

Thanks!

lonegunmanb commented on internal/services/containers/kubernetes_cluster_node_pool_resource.go:
```diff
@@ -502,7 +508,12 @@ func resourceKubernetesClusterNodePoolCreate(d *pluginsdk.ResourceData, meta int
 	}
 
-	d.SetId(*read.ID)
+	id, err := parse.NodePoolID(*read.ID)
+	if err != nil {
+		return err
+	}
```
Can't agree more... I'll modify it as your advice!

lonegunmanb commented:
Hi @tombuildsstuff ,
I've removed Get function as your advice, meanwhile I added an extra check for `kubernetes_cluster_id` in `kubernetes_cluster_node_pool_resource`. If user set a `kubernetes_cluster_id` with a different subscription id than the one set in provider configuration, current code cannot discover it, and will use resource group name and cluster name to retrieve cluster information in provider's subscription, if there is a cluster with same resource group name and cluster name exist in provider's subscription, I think it would cause a bug. Maybe an extra check would avoid this?

tombuildsstuff commented:
@lonegunmanb Node Pools have to be in the same Subscription (and Resource Group iirc) as the Cluster - so that's not an issue

lonegunmanb commented:
> @lonegunmanb Node Pools have to be in the same Subscription (and Resource Group iirc) as the Cluster - so that's not an issue

Roger that, I'll remove this check.

stephybun reviewed (approved):
Thanks @lonegunmanb, LGTM!
